File: app/src/main/java/co/uglytruth/hashtag/instagram/InstagramDialog.java
package co.uglytruth.hashtag.instagram;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.webkit.WebView;

public class InstagramDialog extends Dialog {

    private WebView webView;

    public InstagramDialog(@NonNull Context context) {
        super(context);

       setWebView(context);
    }

    private void setWebView(Context context){

        webView = new WebView(context);
        webView.getSettings().setAppCacheEnabled(true);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new InstagramWebClient());
    }

    public InstagramDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
        setWebView(context);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }
}

File: app/src/main/java/co/uglytruth/hashtag/dialog/InstagramDialog.kt
package co.uglytruth.hashtag.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.ViewGroup
import android.webkit.WebView
import android.widget.FrameLayout



/**
 * Created by tjw127 on 1/22/18.
 */
class InstagramDialog(context: Context?) : Dialog(context) {

    val DIMENSIONS_LANDSCAPE = floatArrayOf(460f, 260f)
    val DIMENSIONS_PORTRAIT = floatArrayOf(280f, 420f)

    val MARGIN = 4
    val PADDING = 2

    val mUrl:String = ""

    val webView:WebView? = null






    val FILL = FrameLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT)
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }


}